.github/workflows/sonar-cloud.yml

```diff
@@ -0,0 +1,93 @@
+name: SonarCloud
+
+on:
+  workflow_dispatch:
+  # workflow_run:
+  #   workflows: ["Pre-commit checks"]
+  #   # branches: [main]
+  #   types: [completed]
+
+  # push:
```

Comment:
Shouldn't we also run this on the main branch? Then we need to run it for pushes to main branch like we do it in other workflows.